test/runtests.jl

```diff
@@ -49,6 +49,21 @@ linreg(x::AbstractVecOrMat, y::AbstractVector) = qr!(simplemm(x)) \ y
     @test coef(lm3) == coef(lm4) ≈ [11, 1, 2, 3, 4]
 end
 
+@testset "Linear Model Cooks Distance - refers to PR #368 and issue #414" begin
+    st_df = DataFrame( 
+        Y=[1.3, 2.3, 5.3, 10., 7.2 , 6.3],
+        XA=[1.5, 6.5, 11.5, 19.9, 17.0, 15.5],
+        XB=[1.8, 7.8, 11.8, 20.5, 17.3, 15.8], 
+        W=[0.6, 1.2, 1.2, 0.6, 1.2, 1.2],
+        CooksD=[1.7122291956, 18.983407026, 0.000118078, 0.8470797843, 0.0715921999, 0.1105843157], 
+        CooksDW=[0.7371619575, 30.979165951, 0.0112480511, 0.5168548491, 0.0600217986, 0.1270980264] )
+
+    t_lm = lm(@formula(Y ~ XA + XB), st_df)
+    @test isapprox(st_df.CooksD, cooksdistance(t_lm))
+    t_lm_w = lm(@formula(Y ~ XA + XB), st_df, wts = st_df.W)
+    @test isapprox(st_df.CooksDW, cooksdistance(t_lm_w))
+end
```

Comment:
It would be useful to have a test for a model without an intercept, and for a model with two collinear columns (as we recently found buggy results in that case).

Reply:
Tests amended. I wasn't sure how collinear the variables needed to be. They have a variance inflation factor ≊ 250 in my test; I hope it is alright.

Reply:
Sorry, I meant 100% collinear variables, like `1:10` and `2:11`. The point is just to test that results are correct when a variable is automatically omitted.

Reply:
Good catch! Indeed it doesn't work. 
I think the problem is that the call to ```modelmatrix(obj)``` returns the same matrix no matter if lm/fit was called with dropcollinear=true or false as if the collinear variable is not dropped. And subsequently, the code fails at trying to invert the matrix.
My wishful thinking is that the cook's distance is meant to be called after fitting the model and hence should expect the collinear variable(s) to be dropped already. What do you think @nalimilan ?
I see that Issue #413 is on a similar topic; could the root cause be the same?

Reply:
OK. Throwing an error is better than silently returning incorrect results. Please just turn the collinearity test into `@test_throws SingularException ...`. The error could probably be fixed by subsetting the `X` matrix to exclude columns for which `coef` is exactly zero, but that doesn't have to be fixed immediately.